internal/ethapi/api.go

```diff
@@ -1134,7 +1134,7 @@ func (s *BlockChainAPI) Call(ctx context.Context, args TransactionArgs, blockNrO
 	return result.Return(), result.Err
 }
 
-func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, gasCap uint64) (hexutil.Uint64, error) {
+func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, overrides *StateOverride, gasCap uint64) (hexutil.Uint64, error) {
```

Comment:
prefer to add this arg in the ending

Reply:
Yes this is a must. Otherwise it becomes a backwards-incompatible change.

Reply:
This function is not the actual external api. In the external api it's the last argument. I added it in the same order with `doCall` for this function.